### ISSUE TITLE
The VM's filter always returns the Pools entities regarldess to the query result

### ISSUE THREAD

**Issue by sgratch** [oVirt/ovirt-web-ui#1183]
The Pool entities are always displayed on VMs dashboard, regardless to the filter type/value query used. It seems that the Pool entity is not included on filter calculations.

For example:
![image](https://user-images.githubusercontent.com/18169498/78682454-aabca880-78f6-11ea-8ca0-cdebca97b66f.png)

This is reproduced on ovirt-web-ui 1version 1.6.1-1

**Comment by hstastna** [comment]
Started looking into this and after I apply some filtering, I would expect ALL of the Pool entities to be displayed on VMs dashboard if they are not included in filter calculations. But that's not true, at least for me.

For me only some of them are displayed. And the number of them (when filtering) varies - depending on filter type/value query used, for example:
- originally (no filtering) I can see 5 Pool entities
- with filtering by _Name_ I can see 2 of them - as on the screenshot above
- with filtering by _Status: Running_ I can see 4 Pool entities

So the problem may be more complicated, I will continue investigating.

**Comment by hstastna** [comment]
@sgratch Hi Sharon, could you, please, specify if the scenario is related to admin/non admin user and maybe also some details about user roles? I think this could help us to figure out what should be displayed. Thanks :)

**Comment by sgratch** [comment]
> @sgratch Hi Sharon, could you, please, specify if the scenario is related to admin/non admin user and maybe also some details about user roles? I think this could help us to figure out what should be displayed. Thanks :)

@hstastna it was reproduced to me with admin user, didn't verify that with non-admin.

**Comment by hstastna** [comment]
@sgratch, what I've found also with @sjd78 is that for admin user, Pool entities are **always** displayed, no matter filtering or number of available VMs from a Pool. It's obviously implemented like this in the code. It would be good to check the behavior also for non-admin user: we expect only those Pool entities with at least 1 available VMs to be displayed so the user could take a VM from that Pool.

Another thing I've found is that **Pool entities are included in filter calculations**, for admin user:
![pools](https://user-images.githubusercontent.com/13417815/79875581-271cb480-83ea-11ea-9f80-ca54a1e7bdf6.png)

But if we filter VMs and Pools the way that the query does not match with some Pool entity - such a Pool entity is just displayed (for admin, together with the result of filtering) and not included in the number of _Results_ displayed in the toolbar, so it just seems that Pool entities are excluded in filter calculations:
![vm](https://user-images.githubusercontent.com/13417815/79877576-af9c5480-83ec-11ea-826e-d616779f1dba.png)

Another proof of this:
![pool2](https://user-images.githubusercontent.com/13417815/79877904-19b4f980-83ed-11ea-8989-1860b37b910e.png)
In the picture above, I was filtering by name "pool2", the result was one Pool entity and one VM "pool2-1" which belongs to that pool (yellow). And the remaining Pool entity "pool" (green) is just displayed extra, as I logged in as an admin and as for admins, Pool entities are always displayed.

---

To sum up, I think that the cards are displayed properly, filtering by itself seems to work ok (not 100% sure about this because when I was searching for a name "vm", some of the VMs weren't displayed), but the problem here regarding Pools and filtering is slightly different: confusing number of _Results_ (toolbar) for admin user when filtering. Check the first picture: the number of _Results_ corresponds to the number of cards being displayed (5 _Results_ and 5 cards incl. Pool entities). But in the second and third pictures not. The problem is "always displaying Pool cards" ability in the context of number of _Results_ when filtering.

**Comment by sgratch** [comment]
@hstastna @sjd78 @lwrigh It seems that when working on pools re-design feature, we ignored the VMs filtering part (check https://github.com/oVirt/ovirt-web-ui/issues/195#issuecomment-449481201) and when working on VMs filter, we ignored the pools re-design part (check https://github.com/oVirt/ovirt-web-ui/issues/18#issuecomment-449479982), so we need to decide now how to combine.

IMHO, we need to treat VMs, Pools and Pool VMs entities the same way for all user types and for all Pools states (w/ or w/o available VMs), i.e. the filter calculation should include all entities in dashboard the same way and display only the ones that fit the filtering regardless to user/entity type and status.
Note that the filter label doesn't indicate which entities it applies to and i guess that's the right way and easiest one for the user.
What do you think?

**Comment by hstastna** [comment]
@sgratch I agree with you. Treating all the entities the same way sounds good to me. And I think it fits to my suggestion written earlier - to solve this problem by just removing the ability to always display Pool entities (when being logged as an admin, of course), but only when filtering. Because I think admins may want to see all the Pool entities, but probably they don't need to see them (Pool entities) also when filtering.

**Comment by sgratch** [comment]
@lwrigh WDYT?

**Comment by lwrigh** [comment]
The VM list should only show entities, either VM or pool, that contain the current search term. If the pool name does not contain the search term it should not appear in the filtered list. It feels misleading if it appears and it does not contain the search term. The filter number should only show the VMs and pools that contain the search term.

**Comment by hstastna** [comment]
After some time and some changes, this issue is reproducible again so I'm reopening it.

Related comment: https://github.com/oVirt/ovirt-web-ui/pull/1241#issuecomment-663162285
